javascript/integration-tests/RegisterAddress.spec.ts

```diff
@@ -35,31 +34,7 @@ describe('RegisterAddress', (): void => {
             const unsubscribe = api.tx.creditcoin
                 .registerAddress('Ethereum', wallet.address, ethOwnershipProof(api, wallet, alice.address))
                 .signAndSend(alice, { nonce: -1 }, async ({ dispatchError, events, status }) => {
-                    testUtils.expectNoDispatchError(api, dispatchError);
-
-                    if (status.isInBlock) {
-                        const balancesWithdraw = events.find(({ event: { method, section } }) => {
-                            return section === 'balances' && method === 'Withdraw';
-                        });
-
-                        expect(balancesWithdraw).toBeTruthy();
-
-                        // const accountId = balancesWithdraw.event.data[0].toString();
-                        if (balancesWithdraw) {
-                            const fee = (balancesWithdraw.event.data[1] as Balance).toBigInt();
-
-                            const unsub = await unsubscribe;
```

Comment:
Just a comment, you are awaiting for `unsubscribe`  inside a callback. Won't unsubscribe resolve only after the callback has finished? How is it possible to await on the output if the output depends on the callback being finished? Is this a common practice in typescript?

Reply:
Hmm, IDK. Polkadot docs https://polkadot.js.org/docs/api/start/api.tx.subs show `unsub()` executed from inside the callback function. 

However if I try to do something like that (see last commit) I get failures: 
```
 ● RegisterRepaymentTransfer › fee is min 0.01 CTC
    TypeError: unsubscribe is not a function
      369 |
      370 |             if (unsubscribe) {
    > 371 |                 unsubscribe();
          |                 ^
      372 |                 resolve(fee);
      373 |             } else {
      374 |                 reject(new Error('Subscription failed'));
```


@jfrank-summit , @nathanwhit do you have a better explanation here?

Reply:
So the promise returned by a call like `api.tx.creditcoin.extrinsic(args).signAndSend(<callback>)` resolves right after the extrinsic is sent and the subscription is created. The promise resolves to the function that terminates the event subscription (unsubscribe). The callback is executed after the promise is resolved, so the output doesn't depend on the callback.